### PR TITLE
Live/jenkins

### PR DESCRIPTION
**Pull request by LiveTembiso**
Fixed Jenkinsfile

### CODE REVIEW

**Review by HelenForbes — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -5,19 +5,37 @@ pipeline {
...
+                     includes: '**/*',
+                     reportName: 'Cycax Docs',
+                     reportTitles: 'Cycax Docs report'
+                  ])
+                }
+            }	
+        }
+	stage('Mypy') {
+	    steps {
+		sh (label: 'Run MyPy',
```
Comment (martinslabber) [inline Jenkinsfile:25]: You are mixing styles here, stick with sh, consistency will help to debug these tasks.
```
sh 'mkdir -p reports'
sh 'hatch run lint:typing |& reports/mypy.txt'
```

I suggest you do not call it logs. Small difference but these are record that we process not just logs.
We should be consistent in all the repos. Whatever name we pick we should use it in all repos.

File: Jenkinsfile [hunk 2]
```diff
@@ -5,19 +5,37 @@ pipeline {
...
+			    mkdir -p logs
+			    hatch run lint:typing | tee logs/mypy.log
+			    '''
+		)
+	    }
+
+	    post{
+                always {
+                  recordIssues(tools: [myPy(name: 'MyPy', pattern: 'logs/mypy.log')])
+                  publishHTML([allowMissing: false, alwaysLinkToLastBuild: false, keepAll: false, reportDir: 'logs/site', reportFiles: 'index.html', reportName: 'MyPy Logs', reportTitles: 'Cycax Mypy logs'])
```
Comment (martinslabber) [inline Jenkinsfile:36]: publishHTML does not make sense here?

File: Jenkinsfile [hunk 3]
```diff
@@ -5,19 +5,37 @@ pipeline {
...
+
+	    post {
+      		 success {
+        	  publishHTML([allowMissing: false,
+                     alwaysLinkToLastBuild: false,
+                     keepAll: true,
+                     reportDir: 'docs/site',
+                     reportFiles: 'index.html',
+                     includes: '**/*',
+                     reportName: 'Cycax Docs',
```
Comment (martinslabber) [inline Jenkinsfile:17]: To be pedantic the title is "CyCAx", all caps with small y and x. 
Docs >> docs
The line "reportTitles: 'Cycax Docs report'" seems weird, "Docs report". Lets just say "CyCAx docs"

**Review by martinslabber — COMMENTED** [review]
Not working on Jenkins yet. Lets get it to work first.

File: Jenkinsfile (hunk 2 above)
Comment (LiveTembiso) [reply Jenkinsfile:36]: Yes it does not, I saw it used somewhere but I guess they were generating more than reports in their commands.

**Review by martinslabber — APPROVED** [review]
Nice

**Pull request merged**